src/restarts/restarts_manager.hpp

```diff
@@ -69,25 +74,48 @@ class RestartsManager : public IRestartsManager
     RestartsManager& operator=(RestartsManager&&) = delete;
 
 private:
-    bool needsAction_(double nextTime, double timeStamp, double timeStep)
+    bool needsCadenceAction_(double const nextTime, double const timeStamp,
+                             double const timeStep) const
     {
         // casting to float to truncate double to avoid trailing imprecision
         return static_cast<float>(std::abs(nextTime - timeStamp)) < static_cast<float>(timeStep);
     }
+    bool needsElapsedAction_(double const nextTime, const double /*timeStamp*/,
+                             const double /*timeStep*/) const
+    {
+        auto const current_time = linux_time_now();
+        return current_time > nextTime;
+    }
+
+
+    auto static linux_time_now() { return std::time(NULL); }
```

Comment:
curious: why "linux"?

Reply:
should be "unix" https://www.unixtimestamp.com/